opm/simulators/flow/FlowsContainer.cpp

```diff
@@ -271,32 +271,56 @@ outputRestart(data::Solution& sol)
     };
 
     using Dir = FaceDir::DirEnum;
-    std::vector<DataEntry<Scalar>> entries;
-    addEntry<gasCompIdx>  (entries, "FLRGAS", UnitSystem::measure::rate,                flores_);
-    addEntry<oilCompIdx>  (entries, "FLROIL", UnitSystem::measure::rate,                flores_);
-    addEntry<waterCompIdx>(entries, "FLRWAT", UnitSystem::measure::rate,                flores_);
-    addEntry<gasCompIdx>  (entries, "FLOGAS", UnitSystem::measure::gas_surface_rate,    flows_);
-    addEntry<oilCompIdx>  (entries, "FLOOIL", UnitSystem::measure::liquid_surface_rate, flows_);
-    addEntry<waterCompIdx>(entries, "FLOWAT", UnitSystem::measure::liquid_surface_rate, flows_);
-
-    std::for_each(entries.begin(), entries.end(),
-                  [&doInsert](auto& array)
-                  {
-                      static const auto dirs = std::array{
-                          std::pair{FaceDir::ToIntersectionIndex(Dir::XMinus), "I-"},
-                          std::pair{FaceDir::ToIntersectionIndex(Dir::XPlus), "I+"},
-                          std::pair{FaceDir::ToIntersectionIndex(Dir::YMinus), "J-"},
-                          std::pair{FaceDir::ToIntersectionIndex(Dir::YPlus), "J+"},
-                          std::pair{FaceDir::ToIntersectionIndex(Dir::ZMinus), "K-"},
-                          std::pair{FaceDir::ToIntersectionIndex(Dir::ZPlus), "K+"},
-                      };
-                      const auto& name = std::get<0>(array);
-                      const auto& measure = std::get<1>(array);
-                      auto& value = std::get<2>(array);
-                      for (const auto& [index, postfix] : dirs) {
-                          doInsert(value[index], name + postfix, measure);
-                      }
-                  });
+    if (this->enableFlores_){
```

Comment:
sorry but this fixes nothing. if the arrays are empty nothing is written and arrays are empty if output is not enabled. if that is not the case, the bug is that arrays have been resized. at the very least you could just if just the addEntry calls.

Reply:
In the test I added then they are written to the restart since BFLOW is in the summary. With this then they are not.

Reply:
right. that was stupid of me. we cannot just use array existence since they are dual purpose of course. but then do as i said, just if the addEntry calls, don't duplicate all the code.

Reply:
Sorry for that and thanks for the elegant fix.